### PR TITLE
Raise in extract_season and extract_month if result is None

### PR DESCRIPTION
**Pull request by jvegasbsc**
We found that when a `extract_month` call in a cube that did not have the required month in it ESMValTool fails with an obscure extract trace

```python
  File "/opt/conda/lib/python3.8/site-packages/esmvalcore/_task.py", line 735, in _run_task
    output_files = task.run()
  File "/opt/conda/lib/python3.8/site-packages/esmvalcore/_task.py", line 242, in run
    self.output_files = self._run(input_files)
  File "/opt/conda/lib/python3.8/site-packages/esmvalcore/preprocessor/__init__.py", line 432, in _run
    product.apply(step, self.debug)
  File "/opt/conda/lib/python3.8/site-packages/esmvalcore/preprocessor/__init__.py", line 300, in apply
    self.cubes = preprocess(self.cubes, step, **self.settings[step])
  File "/opt/conda/lib/python3.8/site-packages/esmvalcore/preprocessor/__init__.py", line 253, in preprocess
    items.extend(item)
TypeError: 'NoneType' object is not iterable
```

This is due to the function not failing but returning None instead. This PR fixes this behaviour by raising an exception inside `extract_month` and `extract_season` so we fail with a better message and stack trace

## Before you get started

-   [ ] [☝ Create an issue](https://github.com/ESMValGroup/ESMValCore/issues) to discuss what you are going to do

## Checklist

-   [ ] PR has a descriptive title for the [changelog](https://docs.esmvaltool.org/projects/esmvalcore/en/latest/contributing.html#branches-pull-requests-and-code-review)
-   [ ] Labels are assigned so they can be used in the [changelog](https://docs.esmvaltool.org/projects/esmvalcore/en/latest/contributing.html#branches-pull-requests-and-code-review)
-   [ ] Code follows the [style guide](https://docs.esmvaltool.org/projects/esmvalcore/en/latest/contributing.html#code-style)
-   [ ] [Documentation](https://docs.esmvaltool.org/projects/esmvalcore/en/latest/contributing.html#documentation) is available for new functionality
-   [ ] YAML files pass [`pre-commit`](https://docs.esmvaltool.org/projects/esmvalcore/en/latest/contributing.html#pre-commit) or [`yamllint`](https://docs.esmvaltool.org/projects/esmvalcore/en/latest/community/introduction.html#yaml) checks
-   [ ] [Circle/CI tests pass](https://docs.esmvaltool.org/projects/esmvalcore/en/latest/contributing.html#branches-pull-requests-and-code-review)
-   [ ] [Codacy code quality checks pass](https://docs.esmvaltool.org/projects/esmvalcore/en/latest/contributing.html#branches-pull-requests-and-code-review)
-   [ ] [Documentation builds successfully](https://docs.esmvaltool.org/projects/esmvalcore/en/latest/contributing.html#branches-pull-requests-and-code-review) on readthedocs
-   [ ] [Unit tests](https://docs.esmvaltool.org/projects/esmvalcore/projects/esmvalcore/en/latest/contributing.html#contributing-to-the-esmvalcore-package) are available

If you make backwards incompatible changes to the recipe format:

-   [ ] Update [ESMValTool](https://github.com/esmvalgroup/esmvaltool) and link the pull request(s) in the description

***

To help with the number pull requests:

-   🙏 We kindly ask you to [review](https://docs.esmvaltool.org/en/latest/community/review.html#review-of-pull-requests) two other [open pull requests](https://github.com/ESMValGroup/ESMValTool/pulls) in this repository

### CODE REVIEW

File: esmvalcore/preprocessor/_time.py [hunk 1]
```diff
@@ -177,6 +182,8 @@ def extract_season(cube, season):
     result = cube.extract(iris.Constraint(clim_season=season))
     for coord in coords_to_remove:
         cube.remove_coord(coord)
+    if result is None:
+        raise ValueError(f'Season {season} not present in cube {cube}')
```
Comment (stefsmeets) [inline esmvalcore/preprocessor/_time.py:186]:
```suggestion
    if result is None:
        raise ValueError(f'Season {season!r} not present in cube {cube}')
```

File: esmvalcore/preprocessor/_time.py [hunk 2]
```diff
@@ -194,14 +201,22 @@ def extract_month(cube, month):
...
     if month not in range(1, 13):
         raise ValueError('Please provide a month number between 1 and 12.')
     if not cube.coords('month_number'):
         iris.coord_categorisation.add_month_number(cube,
                                                    'time',
                                                    name='month_number')
-    return cube.extract(iris.Constraint(month_number=month))
+    result = cube.extract(iris.Constraint(month_number=month))
+    if result is None:
+        raise ValueError(f'Month {month} not present in cube {cube}')
```
Comment (stefsmeets) [inline esmvalcore/preprocessor/_time.py:218]:
```suggestion
    if result is None:
        raise ValueError(f'Month {month!r} not present in cube {cube}')
```

**Review by stefsmeets — APPROVED** [review]
Hi @jvegasbsc , nice work! I have also into such issues, which can be a pain to figure out 👍

Just a tiny suggestion from my side to put the parameter variables in quotation marks.

**Review by Peter9192 — APPROVED** [review]
Looks good to me 👍

**Pull request merged**